### PR TITLE
Avoid new data types and simplify logic

### PR DESCRIPTION
**Pull request by afsalthaj**
* Removed TypeAnnotatedValueResult, as it creates confusion
* Move the entire logic of getting type annootated value from function parameters and function results and make json functions implementations one liners.
* From<TypeAnnotatedValue> for Json instead of JsonFunctionResult, as it can be any Json
* Allow borrowing type-annotated-value to obtain its type

### CODE REVIEW

File: wasm-rpc/src/json.rs [hunk 1]
```diff
@@ -12,141 +12,100 @@
...
-            }
-            JsonValue::Object(map)
-        }
-    })
+    TypeAnnotatedValue::from_function_results(values, expected_types)
+        .map(|result| Json::from(result).0)
 }
 
-pub struct JsonFunctionResult(pub serde_json::Value);
+pub struct Json(pub serde_json::Value);
```
Comment (vigoo) [inline wasm-rpc/src/json.rs:56]: Why do we need this wrapper? I think we don't

File: wasm-rpc/src/type_annotated_value.rs [hunk 2]
```diff
@@ -72,6 +73,96 @@ pub enum TypeAnnotatedValue {
 }
 
 impl TypeAnnotatedValue {
+    pub fn from_function_parameters(
```
Comment (vigoo) [inline wasm-rpc/src/type_annotated_value.rs:76]: This is something I did not see in the first PR but there should not be any JSON code in the `typed_annotated_value` module only in `json`. That's an orthogonal feature to having typeinfo.

You can do `features=["typeinfo"]` and then you don't even have a `serde` dependency. So all these functions constructing `TypedAnnotatedValue` from/to `serde_json::Value` should be in `json`.

Comment (afsalthaj) [reply wasm-rpc/src/type_annotated_value.rs:76]: Sure. Had the same doubt, thanks for confirming

Comment (afsalthaj) [reply wasm-rpc/src/type_annotated_value.rs]: Done

**Review by vigoo — APPROVED** [review]

**Review by justcoon — APPROVED** [review]

**Pull request merged**